### ISSUE TITLE
SAST Scan

### ISSUE THREAD

**Issue by nsoutouras** [department-of-veterans-affairs/va.gov-team#35653]
# Description
Static Code Analysis via Sonarqube.  Scanning for security vulnerabilities.

# User Story
**As a** VSP Engineer
**I need** to make sure the code is scanned each time we introduce new code to the branch.
**So that** production code is as secure as we can tighten by having a plan to remediate vulnerabilities found.  

# Acceptance Criteria (Engineer Check off)
- [ ] SAST scan is set up
- [ ] Report posted to Github

# Technical Description

# Deliverables (Product Manager Check off)
- [ ] Report posted to Github

# Expected finish date
- [ ] March 30

**Comment by sylvesterlee1911** [comment]
we just need to put the flipper on for prod. If you want to do the zap test on staging with the login.gov cred you can


and do the sast scan on the repo likely on the 10th after all the code is merged..maybe the 11th

**Comment by sylvesterlee1911** [comment]
Was waiting on stable build to do scan.  Behind schedule of when we would like to do our sast scan.  Adding documentation to the following location.

https://github.com/department-of-veterans-affairs/va.gov-team/tree/master/products/identity/scans/sast

**Comment by nsoutouras** [comment]
1:1 Notes 3/30: No blockers, just time.

**Comment by sylvesterlee1911** [comment]
Troubleshooting.  Sonarqube is unstable.  --------- uninstall java on the machine.

Make sure it is completely removed and remove the variables created on initial configuration.

**Comment by sylvesterlee1911** [comment]
Reinstall java 11 - https://docs.oracle.com/en/java/javase/11/ (note you must have an oracle account in order to download.)

**Comment by sylvesterlee1911** [comment]
Reconfigure Sonarqube - to point to new Java location.  Update path in .conf folder. wrapper file.  Update path for java location and if on windows machine change  the slash from back to forward as this is a unix tool.  (Mac just cut and paste)

**Comment by sylvesterlee1911** [comment]
Sonarqube needs to be upgraded to enterprise edition. so that the reporting will be included.  There is a cnes code that has been downloaded that may have the capability but may need another ticket to implement and test may be a half a day of work.

**Comment by nsoutouras** [comment]
4/6 1:1 - Shifting to in Review.  @joeniquette can you please take a look at this when able?
